cxx/distributions/zero_mean_normal_test.cc

```diff
@@ -0,0 +1,105 @@
+// Apache License, Version 2.0, refer to LICENSE.txt
+
+#define BOOST_TEST_MODULE test Normal
+
+#include "distributions/zero_mean_normal.hh"
+
+#include <boost/test/included/unit_test.hpp>
+
+#include "util_math.hh"
+namespace bm = boost::math;
+namespace tt = boost::test_tools;
+
+BOOST_AUTO_TEST_CASE(simple) {
+  std::mt19937 prng;
+  ZeroMeanNormal nd(&prng);
+
+  nd.incorporate(5.0);
+  nd.incorporate(-2.0);
+  BOOST_TEST(nd.N == 2);
+
+  nd.unincorporate(5.0);
+  nd.incorporate(7.0);
+  BOOST_TEST(nd.N == 2);
+
+  nd.unincorporate(-2.0);
+  BOOST_TEST(nd.N == 1);
+}
+
+BOOST_AUTO_TEST_CASE(no_nan_after_incorporate_unincorporate) {
```

Comment:
Add a test that incorporate / unicorporate update the variance correctly? (see normal_test.cc)?

Reply:
Done.